tsfile/src/test/java/org/apache/iotdb/tsfile/read/ReadBugTest.java

```diff
@@ -0,0 +1,208 @@
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+import org.apache.iotdb.tsfile.common.conf.TSFileDescriptor;
+import org.apache.iotdb.tsfile.encoding.encoder.Encoder;
+import org.apache.iotdb.tsfile.exception.write.WriteProcessException;
+import org.apache.iotdb.tsfile.file.MetaMarker;
+import org.apache.iotdb.tsfile.file.header.ChunkHeader;
+import org.apache.iotdb.tsfile.file.header.PageHeader;
+import org.apache.iotdb.tsfile.file.metadata.ChunkGroupMetaData;
+import org.apache.iotdb.tsfile.file.metadata.TsDeviceMetadata;
+import org.apache.iotdb.tsfile.file.metadata.TsDeviceMetadataIndex;
+import org.apache.iotdb.tsfile.file.metadata.TsFileMetaData;
+import org.apache.iotdb.tsfile.file.metadata.enums.CompressionType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSEncoding;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.read.common.RowRecord;
+import org.apache.iotdb.tsfile.read.expression.QueryExpression;
+import org.apache.iotdb.tsfile.read.query.dataset.QueryDataSet;
+import org.apache.iotdb.tsfile.utils.Pair;
+import org.apache.iotdb.tsfile.utils.RecordUtils;
+import org.apache.iotdb.tsfile.write.TsFileWriter;
+import org.apache.iotdb.tsfile.write.record.TSRecord;
+import org.apache.iotdb.tsfile.write.schema.FileSchema;
+import org.apache.iotdb.tsfile.write.schema.SchemaBuilder;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ReadBugTest {
+  public static final String DELTA_OBJECT_UID = "delta-3312";
+  private String outputDataFile;
+  private String inputDataFile;
+  private static ReadOnlyTsFile roTsFile = null;
+
+  private static int rowCount;
+  private static int chunkGroupSize;
+  private static int pageSize;
+  public final long START_TIMESTAMP = 1480562618000L;
+
+  @Before
+  public void setUp() throws WriteProcessException, IOException, InterruptedException {
+    outputDataFile = "src/test/resources/testTsFile.tsfile";
+    TSFileDescriptor.getInstance().getConfig().timeSeriesEncoder = "TS_2DIFF";
+    inputDataFile = "src/test/resources/perTestInputData";
+
+    rowCount = 1000;
+    chunkGroupSize = 16 * 1024 * 1024;
+    pageSize = 10000;
+    generateSampleInputDataFile();
+    writeToFile();
+
+    TsFileSequenceReader reader = new TsFileSequenceReader(outputDataFile);
+    roTsFile = new ReadOnlyTsFile(reader);
+  }
+
+  private void generateSampleInputDataFile() throws IOException {
+    File file = new File(inputDataFile);
+    if (file.exists()) {
+      file.delete();
+    }
+    file.getParentFile().mkdirs();
+    FileWriter fw = new FileWriter(file);
+
+    long startTime = START_TIMESTAMP;
+    for (int i = 0; i < rowCount; i++) {
+      // write d1
+      String d1 = "d1," + (startTime + i);
+      if (i % 8 == 0) {
+        d1 += ",s4," + "dog" + i;
+      }
+      fw.write(d1 + "\r\n");
```

Comment:
why add "\r\n" to the end of file, "\r\n" is supported on windwos, how about mac/linux?